releasenotes/notes/fix-overwrite-special-gates-drawn-names-0c0585e70fcdb5c6.yaml

```
@@ -0,0 +1,6 @@
+---
+fixes:
+  - |
+    Fixed the QuantumCircuit.id method to allow the passing of a custom label.
+    In addition the TextDrawing has been update to take into account potential custom
+    labels for gates with "special labels": namely "I", "√X" and "√Xdg".
```

This is actually two separate changes, and we could do with two release notes.  The two changes are: a `circuits` feature addition, which is adding `label` to `id`, `sx` and `sxdg`; and a bug fix, which is making `op.label` take priority over special handling.

We can do that as something like


```suggestion
---
features_circuits:
  - |
    The :meth:`~qiskit.circuit.QuantumCircuit.id`, :meth:`~qiskit.circuit.QuantumCircuit.sx`, and :meth:`~qiskit.circuit.QuantumCircuit.sxdg` methods of
    :class:`~qiskit.circuit.QuantumCircuit` now allow the passing of a custom label.
fixes:
  - |
    The text circuit drawer (:meth:`.QuantumCircuit.draw`) will now prioritize
    custom gate labels over any special handling for instructions such as :class:`.IGate`.
```